relay-server/src/envelope/item.rs

```diff
@@ -851,6 +896,17 @@ pub struct ItemHeaders {
     #[serde(default, skip_serializing_if = "is_false")]
     spans_extracted: bool,
 
+    /// The number of spans in the `event.spans` array.
+    ///
+    /// Should never be set except for transaction items.
+    ///
+    /// When a transaction is dropped before spans were extracted from a transaction,
+    /// this number is used to emit correct outcomes for the spans category.
+    ///
+    /// This number does *not* count the transaction itself.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    span_count: Option<usize>,
```

Comment:
At first I made this a `u32` like `item_count`, but it made the rest of the code unergonomic.